File: client/app/src/main/java/com/rh/heji/data/converters/DateConverters.java
package com.rh.heji.data.converters;

import androidx.room.TypeConverter;

import com.blankj.utilcode.util.TimeUtils;

import java.util.Date;

/**
 * Date: 2020/9/20
 * Author: 锅得铁
 * #时间转换 -注意查的时候月份-1 ，存的时候不处理
 */
public class DateConverters {
    @TypeConverter
    public static Date fromTimestamp(Long value) {
        Date date = value == null ? null : TimeUtils.millis2Date(value);
        return date;
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        Long time = date == null ? null : TimeUtils.date2Millis(date);
        return time;
    }
}

File: client/app/src/main/java/com/rh/heji/data/converters/DateConverters.kt
package com.rh.heji.data.converters

import androidx.room.TypeConverter
import com.blankj.utilcode.util.TimeUtils
import com.squareup.moshi.FromJson
import com.squareup.moshi.ToJson
import java.util.*

/**
 * @date: 2020/9/20
 * @author: 锅得铁
 * #时间转换 -注意查的时候月份-1 ，存的时候不处理
 */
object DateConverters {
    const val DB_PATTERN = "yyyy-MM-dd HH:mm:ss"
    const val SHOW_PATTERN = "yyyy-MM-dd HH:mm"
    @FromJson
    @JvmStatic
    @TypeConverter
    fun str2Date(value: String?): Date {
        return if (value == null) Date() else TimeUtils.string2Date(value, DB_PATTERN)
    }
    @ToJson
    @JvmStatic
    @TypeConverter
    fun date2Str(date: Date?): String {
        return if (date == null) "null" else TimeUtils.date2String(date, DB_PATTERN)
    }
}